controllers/imagejob/imagejob_controller.go

```diff
@@ -152,148 +153,188 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	if imageJob.Status.Phase == "" {
-		nodes := &v1.NodeList{}
-		err := r.List(ctx, nodes)
-		if err != nil {
-			return ctrl.Result{}, err
+	switch imageJob.Status.Phase {
+	case "":
+		if err := r.handleNewJob(ctx, clientset, imageJob); err != nil {
+			return ctrl.Result{}, fmt.Errorf("reconcile new: %w", err)
 		}
-
-		imageJob.Status = eraserv1alpha1.ImageJobStatus{
-			Desired:   len(nodes.Items),
-			Succeeded: 0,
-			Failed:    0,
-			Phase:     eraserv1alpha1.PhaseRunning,
+	case eraserv1alpha1.PhaseRunning:
+		if err := r.handleRunningJob(ctx, clientset, imageJob); err != nil {
+			return ctrl.Result{}, fmt.Errorf("reconcile running: %w", err)
 		}
-
-		if err := updateJobStatus(ctx, clientset, imageJob); err != nil {
-			return ctrl.Result{}, err
+	case eraserv1alpha1.PhaseCompleted, eraserv1alpha1.PhaseFailed:
+		if err := r.handleCompletedJob(ctx, imageJob); err != nil {
+			return ctrl.Result{}, fmt.Errorf("reconcile completion: %w", err)
 		}
+	default:
+		return ctrl.Result{}, fmt.Errorf("reconcile: unexpected imagejob phase: %s", imageJob.Status.Phase)
+	}
 
-		for i := range nodes.Items {
-			nodeName := nodes.Items[i].Name
-			runtime := nodes.Items[i].Status.NodeInfo.ContainerRuntimeVersion
-			runtimeName := strings.Split(runtime, ":")[0]
-			mountPath := getMountPath(runtimeName)
-			if mountPath == "" {
-				log.Println("Incompatible runtime on node ", nodeName)
-				continue
-			}
+	// TODO: What to do if imageJob.DeletionTimestamp is set but jobs are not completed?
```

Comment:
can you open an issue for tracking this

Reply:
https://github.com/Azure/eraser/issues/90